internal/app/kdk/up.go

```diff
@@ -20,20 +20,24 @@ import (
 	"github.com/Sirupsen/logrus"
 	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/mount"
 	"github.com/docker/docker/client"
 	"github.com/docker/go-connections/nat"
 	"github.com/spf13/viper"
 )
 
 func Up(ctx context.Context, dockerClient *client.Client, imageCoordinates string, logger logrus.Entry) error {
-	var binds []string
+	var volumes map[string]struct{}
+	var mounts []mount.Mount
+	volumes = make(map[string]struct{})
 
 	for _, bind := range viper.Get("docker.binds").([]interface{}) {
 		configBind := make(map[string]string)
 		for key, value := range bind.(map[interface{}]interface{}) {
 			configBind[key.(string)] = value.(string)
 		}
-		binds = append(binds, configBind["source"]+":"+configBind["target"])
+		mounts = append(mounts, mount.Mount{Type: mount.TypeBind, Source: configBind["source"], Target: configBind["target"]})
+		volumes[configBind["target"]] = struct{}{}
```

Comment:
Why is this sent in as empty?

Reply:
not sure why volumes has an empty struct -> https://github.com/docker/swarm/blob/master/vendor/github.com/docker/docker/api/types/container/config.go#L59

Reply:
One of these is a type `struct{}` and one of these is an instantiation of an empty struct `struct{}{}`